runtime/__init__.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""Interface to the native C++ ExecuTorch runtime.
+
+Example usage:
+.. code-block:: text
+
+    from pathlib import Path
+
+    import torch
+    from executorch.runtime import LoadProgramConfig, Runtime
+
+    et_runtime: Runtime = Runtime.get()
+    program: Program = et_runtime.load_program(
+        Path("/tmp/program.pte"),
+        config=LoadProgramConfig(verification="internal_consistency"),
+    )
+    print("Program methods:", program.method_names)
+    forward: Method = program.load_method("forward")
+
+    inputs = (torch.ones(2, 2), torch.ones(2, 2))
+    outputs = forward.execute(inputs)
+    print(f"Ran forward({inputs})")
+    print(f"  outputs: {outputs}")
+
+Example output:
+.. code-block:: text
+
+    Program methods: ('forward', 'forward2')
+    Ran forward((tensor([[1., 1.],
+            [1., 1.]]), tensor([[1., 1.],
+            [1., 1.]])))
+      outputs: [tensor([[1., 1.],
+            [1., 1.]])]
+"""
+
+import functools
+from collections import defaultdict
+from pathlib import Path
+from types import ModuleType
+from typing import Any, BinaryIO, Optional, Sequence, Union
+
+from executorch.extension.pybindings.portable_lib import (
+    _get_operator_names,
+    ExecuTorchModule,
+    MethodMeta,
+    Verification,
+)
+
+
+class Method:
+    """An ExecuTorch method, loaded from a Program.
+    TODO: This class should be pybind to the C++ counterpart instead of hosting ExecuTorchModule.
```

Comment:
This TODO is more of a comment than a documentation item: users reading the rendered pydoc shouldn't see this.